event_subscription.go

```diff
@@ -0,0 +1,88 @@
+package litefs
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+)
+
+var (
+	ErrClosed           = errors.New("closed EventSubscription")
+	errUnexpectedStatus = errors.New("unexpected status")
+)
+
+// EventSubscription monitors a LiteFS node for published events.
+type EventSubscription interface {
+	// Next attempts to read the next event from the LiteFS node. An error is
+	// returned if the request fails. Calling `Next()` again after an error will
+	// initiate a new HTTP request. ErrClosed is returned if the EventSubscription
+	// is closed while this method is blocking.
+	Next() (*Event, error)
+
+	// Close aborts any in-progress requests to the LiteFS node.
+	Close()
+}
```

Comment:
I think it was better just having the regular struct. I'm not a huge fan of interfaces unless you really need them. I don't see users mocking this out and if they do, it's pretty easy to mock out the actual HTTP endpoint.